app/src/main/java/com/andrew/tasky/agenda/presentation/screens/event_detail/EventDetailViewModel.kt

```diff
@@ -197,39 +202,89 @@ class EventDetailViewModel @Inject constructor(
-            ),
-            endDateAndTime = LocalDateTime.of(
-                selectedEndDate.value,
-                selectedEndTime.value
-            ),
-            reminderTime = selectedReminderTime.value,
-            photos = photos.value,
-            attendees = attendees.value,
-            isCreator = savedStateHandle.get<AgendaItem.Event>("event")?.isCreator ?: true,
-            host = savedStateHandle.get<AgendaItem.Event>("event")?.host,
-            deletedPhotoKeys = emptyList(), // TODO setup deletedPhotosKeys
-            isGoing = savedStateHandle.get<AgendaItem.Event>("event")?.isGoing ?: true
-        )
+
         // viewModelScope gets cancelled as soon as the Fragment is popped from the backstack,
         // so if you pop it right after inserting an element, this coroutine will be cancelled
         // before it can finish inserting the element. With NonCancellable we make sure it's not
         // going to be cancelled.
         viewModelScope.launch {
             withContext(NonCancellable) {
+                var photosDeleted = 0
+                val targetSize = 1000000
+                val eventPhotos = uiEventPhotos.value
+                    .mapNotNull { eventPhoto ->
+                        when (eventPhoto) {
+                            is UiEventPhoto.LocalPhoto -> {
+                                if (eventPhoto.key == null) {
+                                    val compressedByteArray = eventPhoto.bitmap.let { bitmap ->
+                                        BitmapConverters.bitmapToCompressByteArray(
+                                            bitmap = bitmap,
+                                            targetSize = targetSize
+                                        )
+                                    }
+                                    val key = UUID.randomUUID().toString()
+                                    compressedByteArray?.let {
+                                        LocalPhotoDto(
+                                            key = key,
+                                            byteArray = it
+                                        )
+                                    }?.let { repository.saveLocalPhoto(it) }
+                                    if (compressedByteArray != null) {
+                                        EventPhoto.Local()
+                                    } else {
+                                        photosDeleted++
+                                        null
+                                    }
+                                } else {
+                                    EventPhoto.Local(key = eventPhoto.key)
+                                }
+                            }
+                            is UiEventPhoto.RemotePhoto -> {
+                                eventPhoto.remoteEventPhoto
+                            }
+                            else -> {
+                                null
+                            }
```

Comment:
I'd avoid putting side effects in such lambdas like map, filter, etc. A side effect in this case would be something that's not related to mapping, like compressing & saving the local photo. You can quickly miss that this mapper actually does that and it makes the code quite unreadable.
Instead, after mapping, you can simply run a foreach loop in which you compress and save all images, so that these are separate actions.
Also, why do you always save all photos in the cache? Don't you only want to save them if uploading the event failed (due to missing internet for example)? I'd do that in the repository therefore where you know if that succeeded or not (or in the worker, if you use that).